ros2topic/ros2topic/verb/pub.py

```diff
@@ -90,6 +93,12 @@ def publisher(
 
     node = rclpy.create_node(node_name)
 
+    # Clock that has support for ROS time.
+    # TODO(dhood): use sim time if parameter has been set on the node.
+    _clock = ROSClock()
+    _time_source = TimeSource(node=node)
+    _time_source.attach_clock(_clock)
```

Comment:
Instead of creating a new clock please use the clock already available in the node: `clock = node.get_clock()`